src/wp-includes/block-patterns.php

```diff
@@ -328,6 +328,11 @@ function _register_remote_theme_patterns() {
  * @access private
  */
 function _register_theme_block_patterns() {
+	// Avoid registering patterns on the frontend.
+	if ( ! is_admin() && ! wp_is_json_request() ) {
+		return;
+	}
```

Comment:
If we are doing checks like this, we are we even hooking this function on `init`? We could just call it conditionally in `WP_REST_Block_Patterns_Controller` for example.

Reply:
Yeah, I was thinking about that as well... My concern is that many things in the editor are done in REST-API calls, and a hook like `admin_init` wouldn't be suitable in that case. Do you have any ideas on a hook that would work?

This patch here works when manually testing, I'm trying to figure out how to fix the PHPUnit failures now